s3prl/downstream/runner.py

```diff
@@ -390,3 +410,82 @@ def evaluate(self, split=None, logger=None, global_step=0):
             shutil.rmtree(tempdir)
 
         return [] if type(save_names) is not list else save_names
+
+    def inference(self):
+        filepath = Path(self.args.evaluate_split)
+        assert filepath.is_file()
+        filename = filepath.stem
+
+        if hasattr(self.downstream.model, "load_audio"):
+            wav = self.downstream.model.load_audio(filepath)
+        else:
+            wav, sr = torchaudio.load(str(filepath))
+            assert sr == SAMPLE_RATE
+        wavs = [wav.view(-1).to(self.args.device)]
+
+        for entry in self.all_entries:
+            entry.model.eval()
+
+        with torch.no_grad():
+            features = self.upstream.model(wavs)
+            features = self.featurizer.model(wavs, features)
+            self.downstream.model.inference(features, [filename])
+
+    def push_to_huggingface_hub(self):
+        # Setup auth
+        hf_user = os.environ.get("HF_USERNAME")
+        hf_password = os.environ.get("HF_PASSWORD")
+        huggingface_token = HfApi().login(username=hf_user, password=hf_password)
+        HfFolder.save_token(huggingface_token)
+        print(f"HF Hub user: {hf_user}")
```

Comment:
i wasn't sure how to access the logger so opted for simple `print` statements instead ...

Reply:
Sure no problem.